Readme.md

```diff
@@ -1,65 +1,118 @@
+        for (int i = 0; i < files.Length; i++) {
+            if (files[i].IsValid && !string.IsNullOrWhiteSpace(files[i].FileName)) {
+                string resultFilePath = "~/Content/" + files[i].FileName;
+                files[i].SaveAs(System.Web.HttpContext.Current.Request.MapPath(resultFilePath)); 
+                string file = string.Format("{0} ({1}KB)", files[i].FileName, files[i].ContentLength / 1024);
+                string url = ((IUrlResolutionService)sender).ResolveClientUrl(resultFilePath);
+                e.CallbackData += file + "|" + url + "|";
+            }
+        }
+    }
+    ```
+8. Handle the Upload Control's client-side [FilesUploadComplete](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxUploadControl.FilesUploadComplete) event. The [callbackData](https://docs.devexpress.com/AspNet/js-ASPxClientUploadControlFilesUploadCompleteEventArgs.callbackData) event argument contains the string passed from the server. Get file names and URLs from the string and use them to display links to the uploaded files in the Round Panel:
+
+    ```js
+    function OnFilesUploadComplete(s, e) {
+        var data = e.callbackData.split('|');
+        for (var i = 0; i < data.length; i += 2) {
+            var file = data[i];
+            var url = data[i + 1];
+            var link = document.createElement('A');
+            link.innerHTML = file;
+            link.setAttribute('href', url);
+            link.setAttribute('target', '_blank');
+            var fileContainer = document.getElementById('fileContainer');
+            fileContainer.appendChild(link);
+            fileContainer.appendChild(document.createElement('BR'));
+        }
+    }
+    ```
+
+## Files to Review
+
+* [HomeController.cs](./CS/UploadControlExample/Controllers/HomeController.cs) (VB: [HomeController.vb](./VB/UploadControlExample/Controllers/HomeController.vb))
+* [UploadControlValidationSettings.cs](./CS/UploadControlExample/Models/UploadControlValidationSettings.cs) (VB: [UploadControlValidationSettings.vb](./VB/UploadControlExample/Models/UploadControlValidationSettings.vb))
+* [Index.cshtml](./CS/UploadControlExample/Views/Home/Index.cshtml) (VB: [Index.vbhtml](./VB/UploadControlExample/Views/Home/Index.vbhtml))
+
+## Documentation
 
-![upload control](./images/uploadControl.gif "UploadControl")
+* [Upload Modes](https://docs.devexpress.com/AspNet/9886/components/file-management/file-upload/concepts/upload-modes)
 
-**See Also:**
+## More Examples
 
-<a href="https://www.devexpress.com/Support/Center/p/T185980">T185980 - UploadControl with Model Binding Support - Registration Form</a> <br /><a href="https://www.devexpress.com/Support/Center/p/E4381">E4381 - UploadControl - How to upload a file via AJAX</a> 
+* [Upload Control for ASP.NET MVC - How to bind uploaded files to a model](https://github.com/DevExpress-Examples/uploadcontrol-with-model-binding-support-registration-form-t185980)
```

Comment:
These links have the same heading capitalization issues as the current topic. Is it intentional?

Reply:
Yes, we figured out that this is intentional. The question is revoked.